src/main/java/org/ihtsdo/otf/traceabilityservice/service/BranchUtil.java

```diff
@@ -8,4 +13,82 @@ private BranchUtil() {
 	public static boolean isCodeSystemBranch(String branch) {
 		return branch.equals("MAIN") || branch.startsWith("SNOMEDCT-", branch.lastIndexOf("/") + 1);
 	}
+
+	/**
+	 * Return the full branchPath, including the full branchPath of each ancestor in descending order.
+	 *
+	 * @param branchPath Path of branch to process.
+	 * @return The full branchPath, including the full branchPath of each ancestor in descending order.
+	 */
+	public static List<String> getNodes(String branchPath) {
+		return getNodes(branchPath, true);
+	}
```

Comment:
I needed the full path of each ancestor, i.e. `MAIN/SNOMEDCT-XX` and `MAIN/SNOMEDCT-XX/projectA`, so I created this method.

Reply:
@dmcgihtsdo  Nice idea but I think there is other way which is much simpler than this. You could use ArrayDeque and removeLast instead of pop to achieve the same thing.

Reply:
The problem is only adding `MAIN/SNOMED-XX` to the deque instead of both `MAIN` and `MAIN/SNOMEDCT-XX`. In the context of an Extension, `MAIN/SNOMEDCT-XX` is the root level.

Reply:
@dmcgihtsdo It does add MAIN and MAIN/SNOMEDCT-XX to the duque and we can stop processing whenever a code system branch is done. Currently createAncestorDeque() method has implemented as a stack and we can just change it as a queue. I will add the implementation in the ticket for you so that you can test it. Thanks.